x/bep3/genesis.go

```diff
@@ -0,0 +1,51 @@
+package bep3
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/bep3/types"
+)
+
+// InitGenesis initializes the store state from a genesis state.
+func InitGenesis(ctx sdk.Context, keeper Keeper, supplyKeeper types.SupplyKeeper, gs GenesisState) {
+	if err := gs.Validate(); err != nil {
+		panic(fmt.Sprintf("failed to validate %s genesis state: %s", ModuleName, err))
+	}
+
+	// Set each AssetSupply and store total coin count
+	totalAssetSupplyCoins := sdk.NewCoins()
+	for _, asset := range gs.AssetSupplies {
+		keeper.SetAssetSupply(ctx, asset, []byte(asset.Denom))
+		totalAssetSupplyCoins.Add(sdk.NewCoins(asset))
```

Comment:
```suggestion
		totalAssetSupplyCoins = totalAssetSupplyCoins.Add(sdk.NewCoins(asset))
```

I assume you want to update the state of `totalAssetSupplyCoins`

Reply:
Fixed in https://github.com/Kava-Labs/kava/pull/370/commits/aefaf6fdc9582bf6c83b94700b06889d27bdb39b. Updated to make naming conventions as well it make it more readable in https://github.com/Kava-Labs/kava/pull/370/commits/ac92e26999cf32a5fae8296ba92a05b1da0cf357.